File: pkg/pwrgrep/testdata/fixtures/csharp/Concurrency.cs
public class Concurrency
{
    async Task F(Task a, Task b)
    {
        // ruleid: cs-concurrency
        Task.Run(() => Work());
        // ruleid: cs-concurrency
        var both = Task.WhenAll(a, b);
        // ruleid: cs-concurrency
        lock (this)
        {
            Work();
        }
        // ok: cs-concurrency
        Work();
    }

    void Work() { }
}

File: pkg/pwrgrep/testdata/fixtures/python/concurrency.py
import asyncio
import threading


def f(work):
    # ruleid: python-concurrency
    t = threading.Thread(target=work)
    # ruleid: python-concurrency
    lock = threading.Lock()
    # ok: python-concurrency
    t.start()


async def g(coro):
    # ruleid: python-concurrency
    task = asyncio.create_task(coro)
    # ok: python-concurrency
    await task
